### ISSUE TITLE
Explore options for informing users about errors before they submit

### ISSUE THREAD

**Issue by lauraponce** [18F/cms-hitech-apd#2092]
Since we are allowing the app to save even when there are errors present, there is a need for an error check that lets users see any errors and fix them before submitting. The validation check won’t be on autosave or typing or anything annoying, but will be handled by inline validation as outlined in the design system. 

#2054 and the associated Mural shows the validation requirements on the fields within the app.
https://app.mural.co/t/eapd5455/m/eapd5455/1588781600846/74a95a2db5d0b3e72c73cc7a96e6b501f40dcbd1

### This task is done when...
 - [X] Design patterns for this have been reviewed and considered
 - [x] Add assurances and compliance page to the mural for posterity and info on validating the yes/no questions. 
 - [x] Determine if a field can have a generic error, or if it needs something specific customization 
 - [x] Note customization content language on the Mural 

 - [x] Don’t write validation errors for things that are bugs in the queue to be fixed (call these out on the mural by issue #) 
 - [x] Give developers direction around what the design system says about the validation we want to use. Show different areas of the design system where this guidance exists https://design.cms.gov/components/form-label/

 - [x] Mock-up has been designed (if needed)
 - [x] A prototype is created to show interactions with the help drawer and moving to sections of the app to fix errors 
 - [x] Product has provided input
 - [x] A decision has been made about whether to move forward with implementing or not

Validation check is held for a future ticket as part of the update experience - 
 - [ ] Determine and design where folks will hit a button of some kind that triggers the validation check (may be blocked by content nav issues). Should this be before summaries? Before the executive summary? When will they know they have an error and how fixing it will effect the tables they've already seen. Could end up revisiting #2167 through this task?

**Comment by lauraponce** [comment]
From #2106, this issue should also cover the use case for showing an error message and then coming back to the page later. Currently, the error disappears, but it might make sense to keep it visible.

**Comment by beparticular** [comment]
Mural of errors https://app.mural.co/t/eapd5455/m/eapd5455/1588781600846/74a95a2db5d0b3e72c73cc7a96e6b501f40dcbd1

**Comment by jjames521** [comment]
Looking at something like this for the navigation bar.  It includes a number for amount of errors on the page.  The focus style blue bar would turn red to maintain consistency.

![NavErrorVar3.png](https://images.zenhubusercontent.com/5e7151677c78c820a2a11970/b9ae5157-51d5-40b2-a7c4-2487a2320c1b)

**Comment by beparticular** [comment]
@jjames521 awesome! let's dive into this in design sync tomorrow

**Comment by jjames521** [comment]
@beparticular 

Looking at this interaction from the help slider menu component modified to inform on error validation.


![ErrorValidationVar2.png](https://images.zenhubusercontent.com/5e7151677c78c820a2a11970/7031399e-e3b4-41fa-858c-5fbac0151b51)

**Comment by beparticular** [comment]
We have come to a place where we can close this beast! 

Here are the things remaining before close - 
1) @jjames521 will create a new issue for the field level validations that need to happen for fields with formatting or entry issues that can be completed in Cake Layer 1. 
This issue will contain:
-A mockup of what errors should look like
-All the fields that need to be validated, or field types if some are similar (date, phone number, etc).
- What the content should say for each error message 

2) If any bugs remain documented, but not fixed, in the Mural in 2092, create a separate ticket for each bug. 

3) Required field validations will be added to the roadmap to be revisited in a future Cake Layer around Submitting an APD. 

4) @jeromeleecms  The new Mural link is needed 

@pphillips-fearless can I get a double check on these next steps to see that we have everything here we need for now?

**Comment by jjames521** [comment]
Clickable-

https://www.figma.com/proto/6eVvo7JjvXiovGTR4BoVgK/CMS-eAPD-pages-LP-March2020?node-id=724%3A177&viewport=5725%2C-759%2C0.23146726191043854&scaling=contain

**Comment by jjames521** [comment]
Error Validation Mural Link 

https://app.mural.co/t/cmsmural6360/m/eapd5455/1588781600846/74a95a2db5d0b3e72c73cc7a96e6b501f40dcbd1

**Comment by pphillips-fearless** [comment]
Confirmed -- that sounds like what we discussed.
